README.md

```diff
@@ -1 +1,37 @@
 
+For this example ill be using Management Services, which is a logical group of services that are deployed on management hubs (Previously known as Management Hub Components).
+
+2 Repos will coexist:
+- Preprod Management Service Set
+- Prod Management Service Set
+
+ServiceSet - A repo template for Argo ApplicationSet which deploys a set of services defined in values.yaml to a specific cluster.
+
+A service is a helm chart reference defined by:
+- serviceName
+- repoURL
+- targetRevision
+- path
+
+Lets say I want to add a feature to a project that is a part of Management Service Set (After application CI has passed successfully).
+I'll create a new branch for the feature in the helm chart and make the desired changes.
+
+Then, to check the project in a preprod environment, I will change the targetRevision of the desired service reference in "Preprod Management Service" Set's values.yaml, and sync `Preprod Management Service Set` in Argo. 
+
+Lets say everything went well and the application is behaving expectedly and integrating well with its environment.
+Now we want to deploy it to production.
+
+All we have to do now is to create a merge request in the stand-alone helm chart from the branch we worked on to `master` (Preferebly squash commits to avoid `Updated *.yaml` garbage commits and give the single commit a meaningful name).
```

Comment:
One exception is when the commits are meaningful and should not be squashed

Reply:
Right, but usually with yamls commits are usually something like `Updated *.yaml`